llms/mlx_lm/utils.py

```diff
@@ -69,20 +74,23 @@ def get_model_path(path_or_hf_repo: str, revision: Optional[str] = None) -> Path
     """
     model_path = Path(path_or_hf_repo)
     if not model_path.exists():
-        model_path = Path(
-            snapshot_download(
-                repo_id=path_or_hf_repo,
-                revision=revision,
-                allow_patterns=[
-                    "*.json",
-                    "*.safetensors",
-                    "*.py",
-                    "tokenizer.model",
-                    "*.tiktoken",
-                    "*.txt",
-                ],
+        try:
+            model_path = Path(
+                snapshot_download(
+                    repo_id=path_or_hf_repo,
+                    revision=revision,
+                    allow_patterns=[
+                        "*.json",
+                        "*.safetensors",
+                        "*.py",
+                        "tokenizer.model",
+                        "*.tiktoken",
+                        "*.txt",
+                    ],
+                )
             )
-        )
+        except RepositoryNotFoundError:
+            raise RepoNotFoundError(f"No local or Hugging Face repository found for path: {path_or_hf_repo}.")
```

Comment:
Couldn't this fail for other reasons? TBH I'm not sure what the goal of this new exception is. Maybe you can clarify the purpose?

Reply:
In the case where the user tries to load a local model with the name `example_model_name` and makes a typo (for example `example_model-name`, the code still throws the huggingface's `RepositoryNotFoundError`error, with the following error message:

```bash
Repository Not Found for url: https://huggingface.co/api/models/llama3-17b-instruc/revision/main.
Please make sure you specified the correct `repo_id` and `repo_type`.
If you are trying to access a private or gated repo, make sure you are authenticated.
```

for the following line of code:
```python
model, tokenizer = load("llama3-17b-instruc")
```
which is not very informative, as the user is trying to access a local model. Not essential by any means, just something i encountered my first time using the package.

Reply:
I see. Maybe it is worth augmenting that error message instead because it's possible they intended to download from HF in which case the message is actually useful.

Something like:

```
Model Not Found for {model_path}.
Please make sure you specified the local path or Hugging Face repo id correctly.
If you are trying to access a private or gated Hugging Face repo, make sure you are authenticated.
```

Reply:
This looks good, are you going to add the edits or should I?

Reply:
Would you mind making them?